compiler/packages/babel-plugin-react-compiler/src/Inference/InferReferenceEffects.ts

```diff
@@ -872,11 +872,31 @@ function inferBlock(
                 reason: new Set([ValueReason.Other]),
                 context: new Set(),
               };
+
+        for (const element of instrValue.elements) {
+          if (element.kind === 'Spread') {
+            state.referenceAndRecordEffects(
+              freezeActions,
+              element.place,
+              Effect.ConditionallyMutate,
+              ValueReason.Other,
+            );
```

Comment:
I wonder if we could address most of the negative consequences of this change by just special-casing array here. Basically `isBuiltInArrayType(element) ? Effect.Capture : Effect.ConditionallyMutate` as the effect. Because spreading doesn't mutate arrays, it only mutates if you have an iterator.

Reply:
Good point, done!